vulcanai2/datasets/tabulardataset.py

```diff
@@ -22,27 +25,27 @@ def __init__(self, data, labelColumn="label", joinColumn=None, indexList=None):
             if not joinColumn:
                 raise RuntimeError("You need to provide a joinColumn if a list of csvs are provided")
             dfs = [pd.read_csv(f) for f in data]
-            self.df = utils.stitch_datasets(dfs, joinColumn, indexList)
+            self.df = utils.stitch_datasets(dfs, joinColumn, indexList) #TODO: switch this to joining multiple datasets??
         elif isinstance(data, pd.DataFrame):
             self.df = data
         else:
             self.df = pd.read_csv(data)
 
         self.labelColumn = labelColumn
-        self.n = self.df.shape([0]) #TODO: delete this cause it won't update and that's dumb cause you're letting people access the dataframe object
+
 
     def __len__(self):
         """
         Denotes the total number of samples.
         """
-        return self.n
+        return self.df.shape[0]
 
     def __getitem__(self, idx):
         """
         Generates one sample of data
         """
-        #TODO: check types, possibly wrong
-        return list(self.df.drop(self.labelColumn, axis=1)[[idx]]), list(self.df[[self.labelColumn]].iloc[[idx]])[0]
+        #TODO: check types
+        return self.df.drop(self.labelColumn, axis=1).iloc[[2]].values.tolist()[0], self.df[[self.labelColumn]].iloc[[idx]].values.tolist()[0]
```

Comment:
What is this doing? seems pretty complicated to have on one line, maybe we should break it up?